### PR TITLE
Make service catalog as a root chart

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Moved Service Catalog from core chart to the root
- Moved Instances-UI, Catalog-UI, Brokers-UI, BUC to the service-catalog-addons subchart of the core chart.
- Aligned the docs
- Submitted the issue about unnecessary helm upgrade step in installer #1609 

I have measured the time within Kyma is installed with SC extracted from the core chart and with an old approach.

**Extracted SC:**
SC installation ~ 1min
Core installation ~ 4-4,5min
Kyma (all charts) installation ~8min

**Old approach** (SC in the core chart):
Core installation ~5-5.5min
Kyma installation ~8min

Both approaches provides comparable installation times :)

**Related issue(s)**

#809

### CODE REVIEW

**Comment by piotrmiskiewicz** [comment]
create a doc with description of custom service catalog installation, see https://github.com/kyma-project/kyma/blob/master/docs/kyma/docs/037-inst-custom-istio.md

File: resources/core/charts/service-catalog-addons/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Service Catalog Addons
+
+## Overview
+
+Service Catalog Addons provides Kyma addons to the [Service Catalog](../service-catalog/README.md).
```
Comment (klaudiagrz) [inline resources/core/charts/service-catalog-addons/README.md]:
```suggestion
The Service Catalog Add-ons provide Kyma add-ons to the [Service Catalog](https://github.com/kyma-project/kyma/blob/master/resources/service-catalog/README.md).
```

File: resources/core/charts/service-catalog-addons/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+These addons consist of new views related to the Service Catalog and the Binding Usage Controller which extends the Service Catalog with the additional logic.
```
Comment (klaudiagrz) [inline resources/core/charts/service-catalog-addons/README.md]:
```suggestion
These add-ons consist of the following items:
* Kyma Console views related to the Service Catalog
* Binding Usage Controller which extends the Service Catalog with the additional logic
```

File: resources/core/charts/service-catalog-addons/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## Overview
+
+Service Catalog Addons provides Kyma addons to the [Service Catalog](../service-catalog/README.md).
+
+These addons consist of new views related to the Service Catalog and the Binding Usage Controller which extends the Service Catalog with the additional logic. 
+
+### Views
+
+The Service Catalog Addons provides the following views to the Kyma Console.
```
Comment (klaudiagrz) [inline resources/core/charts/service-catalog-addons/README.md]:
```suggestion
The Service Catalog Add-ons provide the following views to the Kyma Console:
```

File: resources/core/charts/service-catalog-addons/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+* catalog-ui
```
Comment (klaudiagrz) [inline resources/core/charts/service-catalog-addons/README.md]:
```suggestion
* Catalog UI
```

File: resources/core/charts/service-catalog-addons/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+* instances-ui
```
Comment (klaudiagrz) [inline resources/core/charts/service-catalog-addons/README.md]:
```suggestion
* Instances UI
```

File: resources/core/charts/service-catalog-addons/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+* brokers-ui
```
Comment (klaudiagrz) [inline resources/core/charts/service-catalog-addons/README.md]:
```suggestion
* Brokers UI
```

File: resources/core/charts/service-catalog-addons/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# Service Catalog Addons
+
+## Overview
+
+Service Catalog Addons provides Kyma addons to the [Service Catalog](../service-catalog/README.md).
+
+These addons consist of new views related to the Service Catalog and the Binding Usage Controller which extends the Service Catalog with the additional logic. 
+
+### Views
```
Comment (klaudiagrz) [inline resources/core/charts/service-catalog-addons/README.md]: The section as a whole is a bit short. I wonder whether we need the heading at all. We can just skip it and start with such a sentence:
`These are the Kyma Console views:`

And then we can start straight with the BUC section. WDYT?

File: resources/core/charts/service-catalog-addons/README.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+The Service Catalog Addons provides the following views to the Kyma Console.
+
+* catalog-ui
+* instances-ui
+* brokers-ui
+
+### Binding Usage Controller
+
+The Binding Usage Controller provides a capability to inject Secrets to the given applications. It introduces two CRs to achieve the injection:
```
Comment (klaudiagrz) [inline resources/core/charts/service-catalog-addons/README.md]:
```suggestion
The Binding Usage Controller allows you to inject Secrets to a given application. For this purpose, it introduces two custom resources (CRs):
```

File: resources/core/charts/service-catalog-addons/README.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+* ServiceBindingUsage
+* UsageKind
+
+The Binding Usage Controller chart provides two default UsageKinds to Kyma:
+
+* function-usage-kind.yaml
+* deployment-usage-kind.yaml
```
Comment (klaudiagrz) [inline resources/core/charts/service-catalog-addons/README.md]: Maybe a link would be useful here to send the reader off to the proper documentation?

`For more information on these CRs, go to the {name} directory.`

File: resources/service-catalog/README.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
+# Service Catalog
+
+## Overview
+
+The Service Catalog consists of the following sub-charts:
+
+- `catalog` - A Kubernetes Incubator project which provides a Kubernetes-native workflow for integrating with [Open Service Brokers](https://www.openservicebrokerapi.org/). For more information about the project, see the [official documentation](https://github.com/kubernetes-incubator/service-catalog).
```
Comment (klaudiagrz) [inline resources/service-catalog/README.md]:
```suggestion
- `catalog`which is a Kubernetes Incubator project that provides a Kubernetes-native workflow for integration with the [Open Service Brokers API](https://www.openservicebrokerapi.org/). For more information about the project, see the [official documentation](https://github.com/kubernetes-incubator/service-catalog).
```

File: resources/service-catalog/README.md [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+- `etcd-stateful` - A store for data.
+
+## Details
+
+To learn more about the Service Catalog operations and usage, see the [Overview](../../../../docs/service-catalog/docs/001-overview-service-catalog.md) document.
```
Comment (klaudiagrz) [inline resources/service-catalog/README.md]:
```suggestion
To learn more about the Service Catalog operations and usage, see the [Overview](https://github.com/kyma-project/kyma/blob/master/docs/service-catalog/docs/001-overview-service-catalog.md) document.
```

File: resources/service-catalog/README.md [hunk 12]
```diff
@@ -0,0 +1,12 @@
+# Service Catalog
+
+## Overview
+
+The Service Catalog consists of the following sub-charts:
+
+- `catalog` - A Kubernetes Incubator project which provides a Kubernetes-native workflow for integrating with [Open Service Brokers](https://www.openservicebrokerapi.org/). For more information about the project, see the [official documentation](https://github.com/kubernetes-incubator/service-catalog).
+- `etcd-stateful` - A store for data.
```
Comment (klaudiagrz) [inline resources/service-catalog/README.md]:
```suggestion
- `etcd-stateful` which is a data store.
```

File: docs/kyma/docs/038-inst-custom-service-catalog.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+title: Installation with custom Service Catalog deployment
+type: Installation
+---
+
+You can use Kyma with a custom Service Catalog deployment. To enable such implementation, remove the Service Catalog from the list of components that install with Kyma.
+
+## Prerequisites
+
+- Service Catalog in version v0.1.28 or higher. To check the currently supported version of the Service Catalog, see [this](https://github.com/kyma-project/kyma/tree/master/resources/service-catalog/charts/catalog/values.yaml) file.
+>**NOTE:** Follow [this](https://kubernetes.io/docs/tasks/service-catalog/) quick start guide to learn how to install and configure Service Catalog on a Kubernetes cluster.
```
Comment (klaudiagrz) [inline docs/kyma/docs/038-inst-custom-service-catalog.md]:
```suggestion
>**NOTE:** Follow [this](https://kubernetes.io/docs/tasks/service-catalog/) guide to learn how to install and configure the Service Catalog on a Kubernetes cluster.
```

File: resources/core/charts/service-catalog-addons/README.md [hunk 14]
```diff
@@ -0,0 +1,31 @@
+# Service Catalog Add-ons
+
+## Overview
+
+The Service Catalog Add-ons provide Kyma add-ons to the [Service Catalog](https://github.com/kyma-project/kyma/blob/master/resources/service-catalog/README.md).
```
Comment (klaudiagrz) [inline resources/core/charts/service-catalog-addons/README.md]:
```suggestion
The Service Catalog Add-ons provide Kyma extensions to the [Service Catalog](https://github.com/kyma-project/kyma/blob/master/resources/service-catalog/README.md).
```

**Pull request merged**